src/main/java/seedu/address/logic/parser/NameContainsKeywordsPredicateConverter.java

```
@@ -1,24 +1,33 @@
 package seedu.address.logic.parser;
 
-import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
-
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Stack;
 
 import picocli.CommandLine;
+import seedu.address.model.person.Name;
 import seedu.address.model.person.NameContainsKeywordsPredicate;
 
 /**
  * Converter from {@code String} to {@code NameContainsKeywordsPredicate}.
  */
 public class NameContainsKeywordsPredicateConverter implements
-        CommandLine.ITypeConverter<NameContainsKeywordsPredicate> {
+        CommandLine.IParameterConsumer {
+
+    public static final String VALIDATION_REGEX = Name.VALIDATION_REGEX;
     @Override
-    public NameContainsKeywordsPredicate convert(String value) throws Exception {
-        String trimmedArgs = value.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new CommandLine.TypeConversionException(MESSAGE_INVALID_COMMAND_FORMAT);
+    public void consumeParameters(Stack<String> args, CommandLine.Model.ArgSpec argSpec,
+                                  CommandLine.Model.CommandSpec commandSpec) {
+        List<String> keywords = new ArrayList<>();
+        while (!args.isEmpty()) {
+            String keyword = args.peek();
+            if (!keyword.matches(VALIDATION_REGEX)) {
+                break;
```

Instead of breaking here, we should throw an error. For consumers use `ParameterException`